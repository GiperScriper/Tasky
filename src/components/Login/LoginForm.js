import React from 'react';
import { Form, Field } from 'react-final-form';
import { renderInputWithError } from '@/helpers/validation';
import { Link } from 'react-router-dom';

import './login.scss';

const loginForm = ({ isPending, onSubmit }) => (
  <div className="login__container">
    <div className="login__title">Tasky</div>
    <div className="login__bg">
      <Form
        onSubmit={onSubmit}
        render={({ handleSubmit, pristine, invalid }) => (
          <form name="loginForm" onSubmit={handleSubmit} className="login__form">
            <Field name="login" type="text" placeholder="Login" component={renderInputWithError} />
            <Field name="password" type="password" placeholder="Password" component={renderInputWithError} />
            <button type="submit" className="login__button" disabled={isPending}>
              {isPending ? (
                <div className="dotted-spinner">
                  loading
                  <div className="dotted-spinner__item dotted-spinner__item--first" />
                  <div className="dotted-spinner__item dotted-spinner__item--second" />
                  <div className="dotted-spinner__item dotted-spinner__item--third" />
                </div>
              ) : (
                'Sign In'
              )}
            </button>
            <Link to="/register" className="login__register--link">
              Sign Up
            </Link>
          </form>
        )}
      />
    </div>
  </div>
);

export default loginForm;
