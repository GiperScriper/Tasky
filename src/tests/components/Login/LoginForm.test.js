import React from 'react';
import { mount } from 'enzyme';
import LoginForm from '@/components/Login/LoginForm';
import { MemoryRouter as Router } from 'react-router-dom';

describe('<LoginForm />', () => {
  it('should render LoginForm component correctly', () => {
    const props = {
      isPending: false,
      onSubmit: jest.fn(),
    };

    const wrapper = mount(
      <Router>
        <LoginForm {...props} />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
