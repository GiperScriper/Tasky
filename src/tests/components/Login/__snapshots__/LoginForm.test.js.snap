// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render LoginForm component correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "0c7age",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "0c7age",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <loginForm
      isPending={false}
      onSubmit={[Function]}
    >
      <div
        className="login__container"
      >
        <div
          className="login__title"
        >
          Tasky
        </div>
        <div
          className="login__bg"
        >
          <ReactFinalForm
            onSubmit={[Function]}
            render={[Function]}
          >
            <form
              className="login__form"
              name="loginForm"
              onSubmit={[Function]}
            >
              <Field
                component={[Function]}
                format={[Function]}
                name="login"
                parse={[Function]}
                placeholder="Login"
                type="text"
              >
                <Component
                  input={
                    Object {
                      "name": "login",
                      "onBlur": [Function],
                      "onChange": [Function],
                      "onFocus": [Function],
                      "value": "",
                    }
                  }
                  meta={
                    Object {
                      "active": false,
                      "data": Object {},
                      "dirty": false,
                      "dirtySinceLastSubmit": false,
                      "error": undefined,
                      "initial": undefined,
                      "invalid": false,
                      "pristine": true,
                      "submitError": undefined,
                      "submitFailed": false,
                      "submitSucceeded": false,
                      "touched": false,
                      "valid": true,
                      "visited": false,
                    }
                  }
                  placeholder="Login"
                  type="text"
                >
                  <div
                    className="login__item login__item--text"
                  >
                    <input
                      className="login__input"
                      name="login"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Login"
                      spellCheck="false"
                      type="text"
                      value=""
                    />
                  </div>
                </Component>
              </Field>
              <Field
                component={[Function]}
                format={[Function]}
                name="password"
                parse={[Function]}
                placeholder="Password"
                type="password"
              >
                <Component
                  input={
                    Object {
                      "name": "password",
                      "onBlur": [Function],
                      "onChange": [Function],
                      "onFocus": [Function],
                      "value": "",
                    }
                  }
                  meta={
                    Object {
                      "active": false,
                      "data": Object {},
                      "dirty": false,
                      "dirtySinceLastSubmit": false,
                      "error": undefined,
                      "initial": undefined,
                      "invalid": false,
                      "pristine": true,
                      "submitError": undefined,
                      "submitFailed": false,
                      "submitSucceeded": false,
                      "touched": false,
                      "valid": true,
                      "visited": false,
                    }
                  }
                  placeholder="Password"
                  type="password"
                >
                  <div
                    className="login__item login__item--password"
                  >
                    <input
                      className="login__input"
                      name="password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Password"
                      spellCheck="false"
                      type="password"
                      value=""
                    />
                  </div>
                </Component>
              </Field>
              <button
                className="login__button"
                disabled={false}
                type="submit"
              >
                Sign In
              </button>
              <Link
                className="login__register--link"
                replace={false}
                to="/register"
              >
                <a
                  className="login__register--link"
                  href="/register"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </Link>
            </form>
          </ReactFinalForm>
        </div>
      </div>
    </loginForm>
  </Router>
</MemoryRouter>
`;
